python/mlcroissant/mlcroissant/_src/operation_graph/operations/read.py

```diff
@@ -84,9 +85,14 @@ def _read_file_content(self, encoding_format: str, file: Path) -> pd.DataFrame:
         """Extracts the `source` file to `target`."""
         filepath = file.filepath
         if is_git_lfs_file(filepath):
-            download_git_lfs_file(file)
+            download_git_lfs_file(file, node=self.node)
         reading_method = _reading_method(self.node, self.fields)
-        with filepath.open("rb") as file:
+        file = filepath.open("rb")
+        # TODO(https://github.com/mlcommons/croissant/issues/635).
```

Comment:
In the issue, can you add the alternative options?

- Have 1 FileObject to decompress and 1 FileObject for the file
- [Prefered option] Have a new keyword `cr:compression` (similar to https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html)

Reply:
Yes, done! Sorry, I just drafted it to be able to put it in the code :)

Btw I dind't put the first bullet point because I'm not sure how this could easily be implemented in mlcroissant, but feel free to add it to the issue!